src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,873 @@
+		//	for (unsigned int i = 0; i < childIndex; i++)
+		//	{
+		//		TreeViewNode calcNode = node.Children().GetAt(i).as< TreeViewNode > ();
+		//		if (calcNode.IsExpanded())
+		//		{
+		//			allOpenedDescendantsCount += GetExpandedDescendantCount(calcNode);
+		//		}
+		//	}
+
+		//	unsigned int childIndexInFlatTree = GetNextIndexInFlatTree(node) + childIndex + allOpenedDescendantsCount;
+		//	return GetNodeAt(childIndexInFlatTree);
+		//}
+
+		//int ViewModel.CountDescendants(const TreeViewNode& value)
+		//{
+		//	int descendantCount = 0;
+		//	unsigned int size = value.Children().Size();
+		//	for (unsigned int i = 0; i < size; i++)
+		//	{
+		//		var childNode = value.Children().GetAt(i).as< TreeViewNode > ();
+		//		descendantCount++;
+		//		if (childNode.IsExpanded())
+		//		{
+		//			descendantCount = descendantCount + CountDescendants(childNode);
+		//		}
+		//	}
+
+		//	return descendantCount;
+		//}
+
+		private uint IndexOfNextSibling(TreeViewNode childNode)
+		{
+			var parentNode = childNode.Parent;
+			int stopIndex;
+			bool isLastRelativeChild = true;
+			while (parentNode != null && isLastRelativeChild)
+			{
+				int relativeIndex = parentNode.Children.IndexOf(childNode);
+				if (parentNode.Children.Count - 1 != relativeIndex)
+				{
+					isLastRelativeChild = false;
+				}
+				else
+				{
+					childNode = parentNode;
+					parentNode = parentNode.Parent;
+				}
+			}
+
+			if (parentNode != null)
+			{
+				int siblingIndex = parentNode.Children.IndexOf(childNode);
+				var siblingNode = parentNode.Children[siblingIndex + 1];
+				IndexOfNode(siblingNode, out var stopIndex);
+			}
+			else
+			{
+				stopIndex = Size;
+			}
+
+			return stopIndex;
+		}
+
+		private uint GetExpandedDescendantCount(TreeViewNode parentNode)
+		{
+			var allOpenedDescendantsCount = 0;
+			for (var i = 0; i < parentNode.Children.Count; i++)
+			{
+				var childNode = (TreeViewNode)parentNode.Children[i];
+				allOpenedDescendantsCount++;
+				if (childNode.IsExpanded)
+				{
+					allOpenedDescendantsCount += CountDescendants(childNode);
+				}
+			}
+			return allOpenedDescendantsCount;
+		}
+
+		internal bool IsNodeSelected(TreeViewNode targetNode)
+		{
+			return m_selectedNodes.IndexOf(targetNode) != -1;
+		}
+
+		private TreeNodeSelectionState NodeSelectionState(TreeViewNode targetNode)
+		{
+			return targetNode.SelectionState;
+		}
+
+		private void UpdateNodeSelection(TreeViewNode selectNode, TreeNodeSelectionState selectionState)
+		{
+			var node = selectNode;
+			if (selectionState != node.SelectionState)
+			{
+				node.SelectionState = selectionState;
+				var selectedNodes = m_selectedNodes;
+				switch (selectionState)
+				{
+					case TreeNodeSelectionState.Selected:
+						selectedNodes.InsertAtCore(selectedNodes.Count, selectNode);
+						m_selectedNodeChildrenChangedEventTokenVector.push_back(get_self<TreeViewNode>(selectNode).ChildrenChanged({ this, &ViewModel.SelectedNodeChildrenChanged }));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)